Code/Mantid/Framework/API/src/Projection.cpp

```diff
@@ -38,6 +40,64 @@ Projection::Projection(const V3D &u, const V3D &v, const V3D &w) {
   }
 }
 
+Projection::Projection(ITableWorkspace_const_sptr ws) {
```

Comment:
I considered that, but because when workspaces are handled in Mantid they're usually shared pointers, it seemed nicer just to accept a shared pointer than to require the caller to check their shared pointer, dereference it, and then pass a const reference manually.